src/app/logic/Players/ComputerPlayer/ComputerPlayer.js

```diff
@@ -0,0 +1,26 @@
+function createPlayer() {
+    const player = {
+        giveAnswerCallbacks: [],
+
+        onGiveAnswerDo(callback) {
+            this.giveAnswerCallbacks.push(callback);
+        },
+
+        answerQuestion(question) {
+            const answer = this._giveRandomAnswer(question.answers);
+            this.giveAnswerCallbacks.forEach((callback) => {callback(answer, question)});
```

Comment:
Are you sure that every callback will take answer and question as arguments?

Reply:
I think that every callback may take answer and question as arguments.
So this implementation is fine to me.

Reply:
It doesn't have to use answer and question as arguments but you will be able to if you want, as Robert said.